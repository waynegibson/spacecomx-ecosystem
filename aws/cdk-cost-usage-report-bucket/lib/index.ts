export * from './stacks/cost-usage-report-bucket-stack'
