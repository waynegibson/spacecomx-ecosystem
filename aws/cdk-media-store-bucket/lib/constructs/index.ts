export * from './cdn-provider-access-policy'
