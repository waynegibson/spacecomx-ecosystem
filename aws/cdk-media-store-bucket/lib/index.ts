export * from './stacks/media-store-bucket-stack'
