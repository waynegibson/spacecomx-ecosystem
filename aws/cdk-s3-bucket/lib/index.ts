export * from './s3-bucket'
