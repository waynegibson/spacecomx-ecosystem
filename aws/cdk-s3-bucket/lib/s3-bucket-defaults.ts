import { RemovalPolicy } from 'aws-cdk-lib'
import * as s3 from 'aws-cdk-lib/aws-s3'

/**
 * @internal
 */
export function DefaultS3Props (loggingBucket?: s3.Bucket, lifecycleRules?: s3.LifecycleRule[]): s3.BucketProps {
  return {
    versioned: true,
    enforceSSL: true,
    encryption: s3.BucketEncryption.S3_MANAGED,
    blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
    autoDeleteObjects: false,
    removalPolicy: RemovalPolicy.RETAIN,
    lifecycleRules: lifecycleRules ?? [],
    serverAccessLogsBucket: loggingBucket
  }
}
