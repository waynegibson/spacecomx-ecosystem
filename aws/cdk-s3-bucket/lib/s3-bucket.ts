import { Construct } from 'constructs'
import type * as s3 from 'aws-cdk-lib/aws-s3'
import { Bucket } from 'aws-cdk-lib/aws-s3'
import { consolidateProps } from '@spacecomx/cdk-utils'
import { DefaultS3Props } from './s3-bucket-defaults'

export interface S3BucketProps {
  /**
   * User provided props to override the default props for the S3 Bucket.
   *
   * @default - Default props are used
   */
  readonly bucketProps?: s3.BucketProps
}

export class S3Bucket extends Construct {
  public readonly s3Bucket: s3.Bucket

  constructor(scope: Construct, id: string, props: S3BucketProps = {}) {
    super(scope, id)

    const combinedBucketProps = consolidateProps(DefaultS3Props(), props.bucketProps)
    const s3BucketResponse = new Bucket(scope, 'S3Bucket', combinedBucketProps)

    this.s3Bucket = s3BucketResponse
  }
}
