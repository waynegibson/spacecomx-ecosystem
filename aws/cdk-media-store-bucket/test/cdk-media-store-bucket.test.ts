import * as cdk from 'aws-cdk-lib'
import { Template } from 'aws-cdk-lib/assertions'
import { MediaStoreBucketStack } from '../lib'

describe('Media store stack -', () => {
  test('it should create a media storage bucket for development.', () => {
    const app = new cdk.App()

    const stack = new MediaStoreBucketStack(app, 'test-media-store-bucket-stack', {
      environment: 'dev',
    })

    const t = Template.fromStack(stack)

    expect(t).toMatchSnapshot()

    t.hasResourceProperties('AWS::S3::Bucket', {
      BucketEncryption: {
        ServerSideEncryptionConfiguration: [
          {
            ServerSideEncryptionByDefault: {
              SSEAlgorithm: 'AES256',
            },
          },
        ],
      },
      PublicAccessBlockConfiguration: {
        BlockPublicAcls: true,
        BlockPublicPolicy: true,
        IgnorePublicAcls: true,
        RestrictPublicBuckets: true,
      },
    })

    t.hasResource('AWS::S3::Bucket', {
      DeletionPolicy: 'Delete',
      UpdateReplacePolicy: 'Delete',
    })

    t.hasResourceProperties('AWS::Lambda::Function', {
      Description: {
        'Fn::Join': [
          '',
          [
            'Lambda function for auto-deleting objects in ',
            {
              Ref: 'S3Bucket07682993',
            },
            ' S3 bucket.',
          ],
        ],
      },
    })
  })

  test('it should create a media storage bucket for production.', () => {
    const app = new cdk.App()

    const stack = new MediaStoreBucketStack(app, 'test-media-store-bucket-stack', {
      environment: 'prod',
    })

    const t = Template.fromStack(stack)

    expect(t).toMatchSnapshot()

    t.hasResourceProperties('AWS::S3::Bucket', {
      BucketEncryption: {
        ServerSideEncryptionConfiguration: [
          {
            ServerSideEncryptionByDefault: {
              SSEAlgorithm: 'AES256',
            },
          },
        ],
      },
      PublicAccessBlockConfiguration: {
        BlockPublicAcls: true,
        BlockPublicPolicy: true,
        IgnorePublicAcls: true,
        RestrictPublicBuckets: true,
      },
    })

    t.hasResource('AWS::S3::Bucket', {
      DeletionPolicy: 'Retain',
      UpdateReplacePolicy: 'Retain',
    })
  })
})
