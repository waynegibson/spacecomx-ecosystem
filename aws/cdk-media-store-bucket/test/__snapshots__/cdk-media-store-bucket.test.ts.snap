// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media store stack - it should create a media storage bucket for development. 1`] = `
{
  "Outputs": {
    "CDNProviderGroupArn": {
      "Value": {
        "Fn::GetAtt": [
          "CDNProviderAccessPolicyGroup4F014CFF",
          "Arn",
        ],
      },
    },
    "CDNProviderGroupName": {
      "Value": {
        "Ref": "CDNProviderAccessPolicyGroup4F014CFF",
      },
    },
    "CDNProviderUserArn": {
      "Value": {
        "Fn::GetAtt": [
          "CDNProviderAccessPolicyUser5EDA1471",
          "Arn",
        ],
      },
    },
    "CDNProviderUserName": {
      "Value": {
        "Ref": "CDNProviderAccessPolicyUser5EDA1471",
      },
    },
    "MediaStoreBucketArn": {
      "Value": {
        "Fn::GetAtt": [
          "S3Bucket07682993",
          "Arn",
        ],
      },
    },
    "MediaStoreBucketName": {
      "Value": {
        "Ref": "S3Bucket07682993",
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "CDNProviderAccessPolicy23A6E68D": {
      "Properties": {
        "Groups": [
          {
            "Ref": "CDNProviderAccessPolicyGroup4F014CFF",
          },
        ],
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject",
                "s3:ListBucket",
                "s3:GetBucketLocation",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "S3Bucket07682993",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "S3Bucket07682993",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CDNProviderAccessPolicy23A6E68D",
      },
      "Type": "AWS::IAM::Policy",
    },
    "CDNProviderAccessPolicyGroup4F014CFF": {
      "Type": "AWS::IAM::Group",
    },
    "CDNProviderAccessPolicyUser5EDA1471": {
      "Properties": {
        "Groups": [
          {
            "Ref": "CDNProviderAccessPolicyGroup4F014CFF",
          },
        ],
      },
      "Type": "AWS::IAM::User",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F": {
      "DependsOn": [
        "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "5194f926380567b6fdffc61629afdad2afc818614cb227df45e499c69db163a7.zip",
        },
        "Description": {
          "Fn::Join": [
            "",
            [
              "Lambda function for auto-deleting objects in ",
              {
                "Ref": "S3Bucket07682993",
              },
              " S3 bucket.",
            ],
          ],
        },
        "Handler": "index.handler",
        "MemorySize": 128,
        "Role": {
          "Fn::GetAtt": [
            "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Sub": "arn:\${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "S3Bucket07682993": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
        "Tags": [
          {
            "Key": "aws-cdk:auto-delete-objects",
            "Value": "true",
          },
        ],
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "S3BucketAutoDeleteObjectsCustomResource7735AB63": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "S3BucketPolicyF560589A",
      ],
      "Properties": {
        "BucketName": {
          "Ref": "S3Bucket07682993",
        },
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F",
            "Arn",
          ],
        },
      },
      "Type": "Custom::S3AutoDeleteObjects",
      "UpdateReplacePolicy": "Delete",
    },
    "S3BucketPolicyF560589A": {
      "Properties": {
        "Bucket": {
          "Ref": "S3Bucket07682993",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "s3:*",
              "Condition": {
                "Bool": {
                  "aws:SecureTransport": "false",
                },
              },
              "Effect": "Deny",
              "Principal": {
                "AWS": "*",
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "S3Bucket07682993",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "S3Bucket07682993",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
              ],
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::GetAtt": [
                    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
                    "Arn",
                  ],
                },
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "S3Bucket07682993",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "S3Bucket07682993",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;

exports[`Media store stack - it should create a media storage bucket for production. 1`] = `
{
  "Outputs": {
    "CDNProviderGroupArn": {
      "Value": {
        "Fn::GetAtt": [
          "CDNProviderAccessPolicyGroup4F014CFF",
          "Arn",
        ],
      },
    },
    "CDNProviderGroupName": {
      "Value": {
        "Ref": "CDNProviderAccessPolicyGroup4F014CFF",
      },
    },
    "CDNProviderUserArn": {
      "Value": {
        "Fn::GetAtt": [
          "CDNProviderAccessPolicyUser5EDA1471",
          "Arn",
        ],
      },
    },
    "CDNProviderUserName": {
      "Value": {
        "Ref": "CDNProviderAccessPolicyUser5EDA1471",
      },
    },
    "MediaStoreBucketArn": {
      "Value": {
        "Fn::GetAtt": [
          "S3Bucket07682993",
          "Arn",
        ],
      },
    },
    "MediaStoreBucketName": {
      "Value": {
        "Ref": "S3Bucket07682993",
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "CDNProviderAccessPolicy23A6E68D": {
      "Properties": {
        "Groups": [
          {
            "Ref": "CDNProviderAccessPolicyGroup4F014CFF",
          },
        ],
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject",
                "s3:ListBucket",
                "s3:GetBucketLocation",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "S3Bucket07682993",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "S3Bucket07682993",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CDNProviderAccessPolicy23A6E68D",
      },
      "Type": "AWS::IAM::Policy",
    },
    "CDNProviderAccessPolicyGroup4F014CFF": {
      "Type": "AWS::IAM::Group",
    },
    "CDNProviderAccessPolicyUser5EDA1471": {
      "Properties": {
        "Groups": [
          {
            "Ref": "CDNProviderAccessPolicyGroup4F014CFF",
          },
        ],
      },
      "Type": "AWS::IAM::User",
    },
    "S3Bucket07682993": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "S3BucketPolicyF560589A": {
      "Properties": {
        "Bucket": {
          "Ref": "S3Bucket07682993",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "s3:*",
              "Condition": {
                "Bool": {
                  "aws:SecureTransport": "false",
                },
              },
              "Effect": "Deny",
              "Principal": {
                "AWS": "*",
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "S3Bucket07682993",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "S3Bucket07682993",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
